### PR TITLE
enhance: Add background rectangle in Vector.coordinate_label based on keyword arg

### PR DESCRIPTION
**Pull request by Rudransh24**
## Motivation and Explanation: Why and how do your changes improve the library?
In some discussions and PR #1407 it was mentioned how this small enhancement is required in order to give the user an option to be able to display background rectangle or not. 

## Links to added or changed documentation pages
 
Earlier, when Vector.coordinate_label was called, it did not have a parameter for showing background color of the coordinate labels. I have added a parameter to the coordinate_label function which enables user to provide a background color to the coordinates

Old:
```
class A(Scene):
    def construct(self):
        vect = Vector([1, 2])
        label = vect.coordinate_label()
        self.add(vect, label)
```
Screenshot (253)

New:
```
class VectorCoordinateLabel(Scene):
    def construct(self):
        plane = NumberPlane()

        vect_1 = Vector([1, 2])
        vect_2 = Vector([-3, -2])
        label_1 = vect_1.coordinate_label(show_bg_rec=True)
        label_2 = vect_2.coordinate_label(color=YELLOW)

        self.add(plane, vect_1, vect_2, label_1, label_2)
```
Screenshot (248)

## Further Information and Comments
This is my first PR in an open source project (outside of Hacktober Fest). Open to suggestions and modifications.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/geometry.py [hunk 1]
```diff
@@ -1606,6 +1613,9 @@ def construct(self):
             shift_dir -= label.get_right() + DEFAULT_MOBJECT_TO_MOBJECT_BUFFER * RIGHT
         label.shift(shift_dir)
         label.set_color(color)
+        if show_bg_rec:
```
Comment (icedcoffeeee) [inline manim/mobject/geometry.py:1616]: This part is redundant no? Passing `add_background_rectangles_to_entries` in `Matrix` already does these.

File: manim/mobject/geometry.py [hunk 2]
```diff
@@ -1569,6 +1573,8 @@ def coordinate_label(
             The number of dimensions of the vector.
         color
             The color of the label.
+        show_bg_rec
+            Display background rectangle for a number.
 
         Examples
```
Comment (icedcoffeeee) [inline manim/mobject/geometry.py:1579]: The example for the method isn't showing up, as shown [here](https://manimce--1988.org.readthedocs.build/en/1988/reference/manim.mobject.geometry.Vector.html). Could you fix that? The indentation of the manim code block isn't quite right.

**Review by icedcoffeeee — COMMENTED** [review]
I glossed over your PR and left some comments

File: manim/mobject/geometry.py (hunk 1 above)
Comment (Rudransh24) [reply manim/mobject/geometry.py]: Oh yes, that sounds better actually. Thanks!
I have made the change in the function.

File: manim/mobject/geometry.py (hunk 2 above)
Comment (Rudransh24) [reply manim/mobject/geometry.py:1579]: I didn't quite get this one. I saw the page and yes the example is missing. But upon checking the indentation in the function docstring and committing, I couldn't pinpoint the exact issue that could be causing the missing example. I tried crosschecking with the other examples present in the script as well. Will try investigating a bit more.

Comment (icedcoffeeee) [reply manim/mobject/geometry.py:1579]: Ah sorry, I misjudged,  the problem was at the `.. manim` keyword at the start of the example, it needs a double colon

Comment (Rudransh24) [reply manim/mobject/geometry.py:1579]: Okay, I tried that as well. Then thought maybe a missing "example" word in the name could be an issue but even now, on refreshing, the example is not getting displayed.

Comment (Rudransh24) [reply manim/mobject/geometry.py:1579]: Oh yes, found it. There was some build issue too due to wrong variable name in the example. Fixed both. Am able to see the example in the link now.

**Comment by huguesdevimeux** [comment]
thanks for the PR ! 

Could you : 
- Put images examples of what your changes do, 
- Add tests. 

Thanks !

**Comment by Rudransh24** [comment]
> thanks for the PR !
> 
> Could you :
> 
> * Put images examples of what your changes do,
> * Add tests.
> 
> Thanks !

Hello @huguesdevimeux 
I have added example and tests. Could you please review it?
Thanks!

**Review by icedcoffeeee — COMMENTED** [review]
I don't think that colored background rectangles is a desired result.

at line 1623, you can make small change:
```py
if show_bg_rect:
    label[1:].set_color(color)
else:
    label.set_color(color)
```

**Comment by Rudransh24** [comment]
> I don't think that colored background rectangles is a desired result.
> 
> at line 1623, you can make small change:
> 
> ```python
> if show_bg_rec:
>     label[1:].set_color(color)
> else:
>     label.set_color(color)
> ```

I made the change but I am still little confused what this is achieving as this does not change anything. Do we only need background color on the digits(which I did in the first commit) or the whole coordinate(which I am doing right now)?

**Comment by icedcoffeeee** [comment]
> which I did in the first commit

Technically, your first commit did bring the desired result. My suggestion of changes was just a mere simplification 😅

**Comment by Rudransh24** [comment]
> > which I did in the first commit
> 
> Technically, your first commit did bring the desired result. My suggestion of changes was just a mere simplification 😅

> Cancel that, why don't you pass include_background_rectangle to Matrix instead of add_background_rectangles_to_entries and adding a BackgroundRectangle separately?

After some observation I found out that there wasn't any need of BackgroundRectangle Function and a separate if-else condition, I think only passing a parameter `show_bg_rec = True` and `add_background_rectangles_to_entries = show_bg_rec` would give us the desired output which is displaying background on the entries of the coordinates.

`label = Matrix(vect, add_background_rectangles_to_entries=show_bg_rec)`
This line gives me the desired output!
Any thoughts?

**Comment by icedcoffeeee** [comment]
> `label = Matrix(vect, add_background_rectangles_to_entries=show_bg_rec)`
This line gives me the desired output!

I don't think so, it still gives a colored background rect under each element. 

Plus, personally when i pass `show_bg_rec`, i'd expect a background rect of the whole matrix rather than just the entries. But it might be different for everyone, we might need another's opinion.

**Comment by Rudransh24** [comment]
> > `label = Matrix(vect, add_background_rectangles_to_entries=show_bg_rec)`
> > This line gives me the desired output!
> 
> I don't think so, it still gives a colored background rect under each element.
> 
> Plus, personally when i pass `show_bg_rec`, i'd expect a background rect of the whole matrix rather than just the entries. But it might be different for everyone, we might need another's opinion.

Ok, so we actually need 2 different parameters one for the entries' background only, and one for the whole matrix.

**Comment by Rudransh24** [comment]
@huguesdevimeux @icedcoffeeee @behackl @hydrobeam any opinions?

**Comment by icedcoffeeee** [comment]
Hi sorry for the delay. I think a good method to customize the matrix would be to let users pass any keyword args as `**kwargs` instead. The coloring issue is best fixed from within the Matrix constructor. Currently, there isn't a way to specify a color to the matrix init. Could you make that possible?

**Pull request closed without merging**